### PR TITLE
[Builder] Add allo.matmul op supported by linalg

### PR DESCRIPTION
**Pull request by Zhichenzzz**
Support tensor matmul nested.
Support const_tensor using.
Test `allo.matul` in `test_parser.py`.

### CODE REVIEW

File: allo/ir/builder.py [hunk 1]
```diff
@@ -903,6 +908,40 @@ def build_Call(ctx, node):
...
 
+    @staticmethod
+    def build_Matmul(ctx, new_args):
+        ip = ctx.get_ip()
+        dtype = ShapedType(new_args[0].type).element_type
+        argAshape = ShapedType(new_args[0].type).shape
+        argBshape = ShapedType(new_args[1].type).shape
+        if len(argAshape) != 2 or len(argBshape) != 2:
+            raise RuntimeError("Only support two 2D matrix multiplication")
+        shape = [argAshape[0], argBshape[1]]
```
Comment (chhzh123) [inline allo/ir/builder.py]: Nit: Use tuple for shape: `(argAshape[0], argBshape[1])`

File: allo/ir/builder.py [hunk 2]
```diff
@@ -903,6 +908,40 @@ def build_Call(ctx, node):
...
+
+        # pylint: disable=unexpected-keyword-arg
+        with ip:
+            memref_type = MemRefType.get(shape, dtype)
+            alloc_op = memref_d.AllocOp(memref_type, [], [], ip=ip)
+            # alloc_op = linalg_d.InitTensorOp(shape, dtype)
```
Comment (chhzh123) [inline allo/ir/builder.py:926]: Add a `TODO` here.
```python
# TODO: Use linalg_d.InitTensorOp when tensor dialect is supported
```

File: tests/test_parser.py [hunk 3]
```diff
@@ -824,6 +825,66 @@ def kernel() -> float32[M, N]:
...
+        C = allo.matmul(A, B)
+        return C
+
+    with pytest.raises(RuntimeError) as excinfo:
+        s = allo.customize(kernel)
+        f = s.build()
+        np_2 = np.zeros((M, K, M, N), dtype="int32")
+        np_2 = f(np_0, np_1)
+        np.testing.assert_array_equal(np_2, np.matmul(np_0, np_1))
+        print(s.module)
```
Comment (chhzh123) [inline tests/test_parser.py]: Delete these few lines. Should be exactly errored out when calling `allo.customize`

Comment (Zhichenzzz) [reply tests/test_parser.py]: ok

**Pull request merged**